### PR TITLE
Feature/peer 4 login via intranet

### PR DESCRIPTION
**Pull request by SharyuMarwadi**

### CODE REVIEW

File: internal/api/user.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+func loginUser(userSvc user.Service) http.HandlerFunc {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		authToken := req.Header.Get(constants.IntranetAuth)
+		if authToken == "" {
+			err := apperrors.InvalidAuthToken
+			dto.ErrorRepsonse(rw, err)
+			return
+		}
+
+		validateResp, err := userSvc.ValidatePeerly(req.Context(), authToken)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/api/user.go]: Done

File: internal/api/validation/user.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	"fmt"
+
+	"github.com/joshsoftware/peerly-backend/internal/pkg/apperrors"
+	"github.com/joshsoftware/peerly-backend/internal/pkg/dto"
+)
+
+func GetIntranetUserDataValidation(user dto.IntranetUserData) (err error) {
+
+	if user.PublicProfile.FirstName == "" || user.PublicProfile.LastName == "" || user.EmpolyeeDetail.Designation.Name == "" || user.Email == "" || user.EmpolyeeDetail.Grade == "" {
+		fmt.Println("Invalid user data")
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/api/validation/user.go]: Done

File: internal/app/users/service.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+type service struct {
+	userRepo repository.UserStorer
+}
+
+type Service interface {
+	ValidatePeerly(ctx context.Context, authToken string) (data dto.ValidateResp, err error)
+	GetIntranetUserData(ctx context.Context, req dto.GetIntranetUserDataReq) (data dto.IntranetUserData, err error)
+	LoginUser(ctx context.Context, u dto.IntranetUserData) (dto.LoginUserResp, error)
+	RegisterUser(ctx context.Context, u dto.IntranetUserData) (user dto.User, err error)
+	GetUserListIntranet(ctx context.Context, reqData dto.GetUserListReq) (data []dto.IntranetUserData, err error)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/app/users/service.go]: Done

File: internal/app/users/service.go [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+func NewService(userRepo repository.UserStorer) Service {
+	return &service{
+		userRepo: userRepo,
+	}
+}
+
+func (us *service) ValidatePeerly(ctx context.Context, authToken string) (data dto.ValidateResp, err error) {
+	client := &http.Client{}
+	validationReq, err := http.NewRequest("POST", "https://pg-stage-intranet.joshsoftware.com/api/peerly/v1/sessions/login", nil)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/app/users/service.go]: Done

File: internal/app/users/service.go [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+		return
+	}
+
+	return
+}
+
+func (us *service) GetIntranetUserData(ctx context.Context, req dto.GetIntranetUserDataReq) (data dto.IntranetUserData, err error) {
+
+	client := &http.Client{}
+	url := fmt.Sprintf("https://pg-stage-intranet.joshsoftware.com/api/peerly/v1/users/%d", req.UserId)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/app/users/service.go]: Done

File: internal/app/users/service.go [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+	intranetReq, err := http.NewRequest("GET", url, nil)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/app/users/service.go]: Done

File: internal/pkg/apperrors/errors.go [hunk 7]
```diff
@@ -49,16 +52,17 @@ func ErrKeyNotSet(key string) (err error) {
...
 func GetHTTPStatusCode(err error) int {
 	switch err {
-	case InternalServerError, JSONParsingErrorResp, InvalidIntranetData:
+	case InternalServerError, JSONParsingErrorResp:
 		return http.StatusInternalServerError
-	case OrganizationNotFound, InvalidOrgId, GradeNotFound:
+	case OrganizationNotFound, InvalidOrgId, GradeNotFound, PageParamNotFound, InvalidIntranetData:
 		return http.StatusNotFound
-	case InvalidId, JSONParsingErrorReq, TextFieldBlank, InvalidCoreValueData, InvalidParentValue, DescFieldBlank, UniqueCoreValue:
+	case InvalidId, JSONParsingErrorReq, TextFieldBlank, InvalidCoreValueData, InvalidParentValue, DescFieldBlank, UniqueCoreValue, RepeatedUser:
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/pkg/apperrors/errors.go]: Done

File: internal/pkg/constants/user.go [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package constants
+
+var AuthorizationHeader string = "Authorization"
+var ClientCode = "Client-Code"
+var UserRole = "user"
+var PerPage = 400
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/pkg/constants/user.go]: Done

File: internal/pkg/dto/users.go [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+	UserId   int    `json:"user_id"`
+}
+
+type GetIntranetUserDataReq struct {
+	Token  string
+	UserId int
+}
+
+type Claims struct {
+	Id   int
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/pkg/dto/users.go]: Done

File: internal/pkg/middleware/jwtMiddleware.go [hunk 10]
```diff
@@ -1,9 +1,68 @@
...
+		jwtKey := config.JWTKey()
+		authToken := req.Header.Get(constants.AuthorizationHeader)
+		if authToken == "" {
+			logger.Error("Empty auth token")
+			err := apperrors.InvalidAuthToken
+			dto.ErrorRepsonse(rw, err)
+			return
+		}
+
+		fmt.Println("authtoken: ", authToken)
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/pkg/middleware/jwtMiddleware.go]: It was for debugging, removed it

File: internal/repository/postgresdb/coreValues.go [hunk 11]
```diff
@@ -52,14 +48,14 @@ func (cs *coreValueStore) ListCoreValues(ctx context.Context) (coreValues []repo
...
-	queryBuilder := sq.
+	queryBuilder := repository.Sq.
 		Select(CoreValueColumns...).
 		From(cs.TableName).
 		Where(squirrel.Eq{"id": coreValueID})
 
 	getCoreValueQuery, args, err := queryBuilder.ToSql()
 	if err != nil {
-		logger.Errorf("error in generating squirrel query, err: %w", err)
+		logger.Errorf("error in generating squirrel query, err: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (SharyuMarwadi) [reply internal/repository/postgresdb/coreValues.go:58]: There are different custom errors for different situations in this function. Hence cannot wrap and return it

File: internal/app/users/service.go [hunk 12]
```diff
@@ -231,24 +239,25 @@ func (us *service) RegisterUser(ctx context.Context, u dto.IntranetUserData) (us
 	return
 }
 
-func (us *service) GetUserListIntranet(ctx context.Context, reqData dto.GetUserListReq) (data []dto.IntranetUserData, err error) {
+func (us *service) ListIntranetUsers(ctx context.Context, reqData dto.GetUserListReq) (data []dto.IntranetUserData, err error) {
 	client := &http.Client{}
-	url := fmt.Sprintf("https://pg-stage-intranet.joshsoftware.com/api/peerly/v1/users?page=%d&per_page=%d", reqData.Page, constants.PerPage)
-	intranetReq, err := http.NewRequest("GET", url, nil)
+	url := config.IntranetBaseUrl() + fmt.Sprintf("/api/peerly/v1/users?page=%d&per_page=%d", reqData.Page, constants.DefaultPageSize)
```
Comment (yama11299) [inline internal/app/users/service.go:244]: constant is defined for this path right? use it here.

File: internal/pkg/constants/system.go [hunk 13]
```diff
@@ -9,4 +9,7 @@ const (
 	DBURI                  = "DB_URI"
 	IntranetClientCode     = "INTRANET_CLIENT_CODE"
 	MigrationFolderPath    = "MIGRATION_FOLDER_PATH"
+	IntanetAuthToken       = "INTRANET_AUTH_TOKEN"
```
Comment (yama11299) [inline internal/pkg/constants/system.go:12]: IntanetAuthToken -> IntranetAuthToken

**Review by yama11299 — CHANGES_REQUESTED** [review]

File: internal/pkg/constants/system.go (hunk 13 above)
Comment (SharyuMarwadi) [reply internal/pkg/constants/system.go]: Done

**Comment by yama11299** [comment]
There are still some small changes but approving for now.

**Comment by SharyuMarwadi** [comment]
Okay

**Pull request merged**